pkg/api/health.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	"time"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+func (hs *HTTPServer) databaseHealthyWithCache() bool {
+	cacheKey := "db-healthy"
+
+	if cached, found := hs.CacheService.Get(cacheKey); found {
+		return cached.(bool)
+	}
+
+	healthy := hs.databaseHealthy()
+
+	hs.CacheService.Set(cacheKey, healthy, time.Second*5)
+	return healthy
+}
+
+func (hs *HTTPServer) databaseHealthy() bool {
+	return bus.Dispatch(&models.GetDBHealthQuery{}) == nil
+}
```

Comment:
Please just inline this method, since it's too simple to extract as a separate function (doesn't outweigh the cost of the abstraction).